source/common/modules/markdown-editor/hooks/codeblock-classes.ts

```diff
@@ -57,93 +59,43 @@ function handleNewline (cm: CodeMirror.Editor, name: string): void {
  * @param   {CodeMirror.Editor}  cm  The instance
  */
 function findCode (cm: CodeMirror.Editor): void {
-  const codeBlockLines = []
   const lineCount = cm.lineCount()
   const codeBlockRE = /^(?:\s{0,3}`{3}|~{3}).*/
-  const indentedRE = /^\s{4,}.*$/
-  const blankishRE = /^\s*$/
 
-  // Check lines for code blocks
-  for (let lineNum = 0; lineNum < lineCount; lineNum++) {
-    let line = cm.getLine(lineNum)
-    // Fenced code found
-    if (codeBlockRE.test(line)) {
-      // Find fenced code end
-      while (lineNum + 1 < lineCount && !codeBlockRE.test(cm.getLine(lineNum + 1))) {
-        codeBlockLines.push(++lineNum)
-      }
-
-      // Skip line that marks fenced code end
-      lineNum++
-    }
-
-    // Possible match for indented code found
-    if (indentedRE.test(line)) {
-      let prevLine = lineNum - 1
+  let inCodeBlock = false
 
-      // Verify match
-      // If this is the first line and either already indented code or prepended by an empty line
-      if (prevLine >= 0 && (codeBlockLines.includes(prevLine) || blankishRE.test(cm.getLine(prevLine)))) {
-        // If this is not preformatted markdown (e.g. a list)
-        if (!cm.getLineTokens(lineNum).some(token => String(token.type).includes('formatting-list'))) {
-          let probeLine = 1
+  cm.startOperation()
 
-          // Skip ahead to the end of the potential code block
-          while (lineNum + probeLine < lineCount && indentedRE.test(cm.getLine(lineNum + probeLine))) {
-            probeLine++
-          }
-
-          // Check if end of file or block appended by empty line, making it a legal code block
-          if (lineNum + probeLine === lineCount || blankishRE.test(cm.getLine(lineNum + probeLine))) {
-            codeBlockLines.push(...range(lineNum, lineNum + probeLine))
-            // Skip ahead until after code block
-            lineNum += probeLine - 1
-          }
-        }
+  // Check lines for code blocks
+  for (let i = 0; i < lineCount; i++) {
+    const line = cm.getLine(i)
+    if (!inCodeBlock) {
+      // Not a codeblock: Remove any class
+      cm.removeLineClass(i, 'wrap', codeblockClass)
+      cm.removeLineClass(i, 'wrap', codeblockClassOpen)
+      cm.removeLineClass(i, 'wrap', codeblockClassClose)
+      if (codeBlockRE.test(line)) {
+        // Begin a codeblock
+        inCodeBlock = true
+        // Increment the lineCount and apply the code start line to that line
+        cm.addLineClass(++i, 'wrap', codeblockClass)
```

Comment:
I noticed that when you add a newline to a codeblock that only had one line of code, the first code line still has the `codeblockClassClose` class. We could fix that by removing that class first:
```suggestion
        cm.removeLineClass(++i, 'wrap', codeblockClassClose)
        cm.addLineClass(i, 'wrap', codeblockClass)
```

Reply:
Implemented!